#! /bin/sh /usr/share/dpatch/dpatch-run
## 24_dont_free_unitialised_pointer.dpatch by  <crimsun@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't free() uninitialised pointer

@DPATCH@

--- hg-alsa-lib-1.0.10/src/conf.c	2005-10-12 12:08:53.000000000 -0400
+++ hg-for-pitti-alsa-lib-1.0.10/src/conf.c	2006-05-06 23:42:05.000000000 -0400
@@ -3808,7 +3808,7 @@
 		snd_config_t *def, *sub, *typ;
 		const char *new = str;
 		const char *tmp;
-		char *val;
+		char *val = NULL;
 		err = parse_arg(&new, &varlen, &val);
 		if (err < 0)
 			goto _err;
